### DISCUSSION TITLE
CuyAnimeList | Next JS App Router

### DISCUSSION THREAD

**Discussion by deaafrizal** [deaafrizal/forum#12] [My Projects]
![image](https://github.com/deaaprizal/forum/assets/43339356/84917d71-b23c-4ef6-8ca3-63288847b7bd)

[free source code](https://github.com/deaaprizal/cuyanimelist)

[tutorial video playlist](https://www.youtube.com/playlist?list=PLc6SEcJkQ6DwSCyM53kqpM1yNyXT80d3U)

nitip 👌🔥

**Comment by someone-only** [comment]
Makasih ya bang dea tutorialnya walaupun beda framework tapi tetep masih bisa ngikutin karna penjelasan dari bang dea mudah dimengerti

**Reply by deaafrizal** [reply]
asoyy gasgasss

**Comment by deaafrizal** [comment]
PRISMA SINGLETON - untuk sesi 10

```
import { PrismaClient } from '@prisma/client';

let prisma;

if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient();
  }
  prisma = global.prisma;
}

export default prisma;
```

**Comment by deaafrizal** [comment]
SESI 10 - upgrade next js ke versi 14:

`npm i next@latest react@latest react-dom@latest eslint-config-next@latest`

**Comment by stheffen** [comment]
gua lagi nyoba sesi 10 di cuyanimelist nah gua nemuin masalah pas 46.08 , di terminal saya munculin ini
 ⨯ [TypeError: The "payload" argument must be 
of type object. Received null] {
  code: 'ERR_INVALID_ARG_TYPE',
  page: '/api/v1/collection'
}
 POST /api/v1/collection 500 in 434ms

apakah ada yang bisa bantu ?

**Reply by antyo-system** [reply]
ctrl + Z
